cpp/tests/structure/graph_test.cpp

```diff
@@ -143,99 +147,254 @@ class Tests_Graph : public ::testing::TestWithParam<Graph_Usecase> {
     ASSERT_EQ(graph_view.get_number_of_vertices(), number_of_vertices);
     ASSERT_EQ(graph_view.get_number_of_edges(), number_of_edges);
 
-    std::vector<edge_t> h_cugraph_offsets(graph_view.get_number_of_vertices() + 1);
-    std::vector<vertex_t> h_cugraph_indices(graph_view.get_number_of_edges());
-    auto h_cugraph_weights =
-      graph.is_weighted() ? std::optional<std::vector<weight_t>>(graph_view.get_number_of_edges())
-                          : std::nullopt;
-
-    raft::update_host(h_cugraph_offsets.data(),
-                      graph_view.get_matrix_partition_view().get_offsets(),
-                      graph_view.get_number_of_vertices() + 1,
-                      handle.get_stream());
-    raft::update_host(h_cugraph_indices.data(),
-                      graph_view.get_matrix_partition_view().get_indices(),
-                      graph_view.get_number_of_edges(),
-                      handle.get_stream());
-    if (h_cugraph_weights) {
-      raft::update_host((*h_cugraph_weights).data(),
-                        *(graph_view.get_matrix_partition_view().get_weights()),
+    if (graph_usecase.check_correctness) {
+      std::vector<edge_t> h_cugraph_offsets(graph_view.get_number_of_vertices() + 1);
+      std::vector<vertex_t> h_cugraph_indices(graph_view.get_number_of_edges());
+      auto h_cugraph_weights =
+        graph.is_weighted() ? std::optional<std::vector<weight_t>>(graph_view.get_number_of_edges())
+                            : std::nullopt;
+
+      raft::update_host(h_cugraph_offsets.data(),
+                        graph_view.get_matrix_partition_view().get_offsets(),
+                        graph_view.get_number_of_vertices() + 1,
+                        handle.get_stream());
+      raft::update_host(h_cugraph_indices.data(),
+                        graph_view.get_matrix_partition_view().get_indices(),
                         graph_view.get_number_of_edges(),
                         handle.get_stream());
-    }
+      if (h_cugraph_weights) {
+        raft::update_host((*h_cugraph_weights).data(),
+                          *(graph_view.get_matrix_partition_view().get_weights()),
+                          graph_view.get_number_of_edges(),
+                          handle.get_stream());
+      }
 
-    CUDA_TRY(cudaStreamSynchronize(handle.get_stream()));
+      CUDA_TRY(cudaStreamSynchronize(handle.get_stream()));
```

Comment:
I assume now we should better use `handle.sync_stream()` with the recent raft update.

Reply:
Will look at this.  raft changes were not merged when I started working on this.

Reply:
We have a number of calls like this in the code base, I'm inclined to address this in a separate PR that picks up all of them.